### ISSUE TITLE
Question: How is the loss being computed?

### ISSUE THREAD

**Issue by tanmaylaud** [jwieting/paraphrastic-representations-at-scale#8]
HI @jwieting,
I am confused how this particular line implements the loss mentioned in the paper?
https://github.com/jwieting/paraphrastic-representations-at-scale/blob/20c66dd01002da5eb9e239e613afc598ecc71b06/models.py#L139

What are g1,g2,p1,p2 ?

**Comment by jwieting** [comment]
Hello! g1g2 is the cosine of the two gold pairs (g1 and g2), and g1p1 is the cosine of the first gold pair (g1) with its negative example (p1). The second term is the same, except the negative pair here is between g2 and p2. The loss is described in more detail here (and the paper): https://pytorch.org/docs/stable/generated/torch.nn.MarginRankingLoss.html, with a margin defined by a hyperparamter (delta).
